### ISSUE TITLE
Add `Outputs` base class

### ISSUE THREAD

**Issue by JosePizarro3** [FAIRmat-NFDI/nomad-simulations#27]
@ndaelman-hu @JFRudzinski 

This is the issue to add the basic `Outputs` section. I will call for review once it is ready.

Some basic ideas:
- It should contain `model_system_ref` (repeats=False), `atoms_state_ref` (repeats=True), `orbitals_state_ref` (repeats=True).
- It should contain `is_derived` and `outputs_ref` for when a property is derived during normalization.
- It should contain `is_converged` and be connected with `numerical_settings.SelfConsistency`

I leave out handling times for another issue.

**Comment by JosePizarro3** [comment]
Actually, I am not sure if we need a repeated section for `atoms_state_ref` or `orbitals_state_ref`. I guess they should not, as each output property will individually refer to each multi-grain system / individual grain / atom / orbital, and not to the whole group of them.

**Comment by JosePizarro3** [comment]
Last:
- We need to include the scf steps within the `Outputs`. I will think on the schema and come with an idea, but I guess we can keep the converged value of each property outside each sfc step, and in the sfc steps just the evolution of such property.

**Comment by ndaelman-hu** [comment]
> Last:
> 
>     * We need to include the scf steps within the `Outputs`. I will think on the schema and come with an idea, but I guess we can keep the converged value of each property outside each sfc step, and in the sfc steps just the evolution of such property.

Yes, that sounds sensible. The issue with scf, is that most people don't use it. It just gives nice stats and convergence plots :p
